x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -178,215 +202,365 @@ export class ExecutionHandler<
 
-        ruleRunMetricsStore.incrementNumberOfGeneratedActionsByConnectorType(actionTypeId);
+      ruleRunMetricsStore.incrementNumberOfGeneratedActionsByConnectorType(actionTypeId);
 
-        if (ruleRunMetricsStore.hasReachedTheExecutableActionsLimit(actionsConfigMap)) {
-          ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
-            actionTypeId,
-            status: ActionsCompletion.PARTIAL,
-          });
+      if (ruleRunMetricsStore.hasReachedTheExecutableActionsLimit(actionsConfigMap)) {
+        ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
+          actionTypeId,
+          status: ActionsCompletion.PARTIAL,
+        });
+
+        logger.debug(
+          `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions has been reached.`
+        );
+
+        break;
+      }
+
+      if (
+        ruleRunMetricsStore.hasReachedTheExecutableActionsLimitByConnectorType({
+          actionTypeId,
+          actionsConfigMap,
+        })
+      ) {
+        if (!ruleRunMetricsStore.hasConnectorTypeReachedTheLimit(actionTypeId)) {
           logger.debug(
-            `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions has been reached.`
+            `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions for connector type ${actionTypeId} has been reached.`
           );
-          break;
         }
 
-        if (
-          ruleRunMetricsStore.hasReachedTheExecutableActionsLimitByConnectorType({
-            actionTypeId,
-            actionsConfigMap,
-          })
-        ) {
-          if (!ruleRunMetricsStore.hasConnectorTypeReachedTheLimit(actionTypeId)) {
-            logger.debug(
-              `Rule "${this.rule.id}" skipped scheduling action "${action.id}" because the maximum number of allowed actions for connector type ${actionTypeId} has been reached.`
-            );
-          }
-          ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
-            actionTypeId,
-            status: ActionsCompletion.PARTIAL,
-          });
-          continue;
+        ruleRunMetricsStore.setTriggeredActionsStatusByConnectorType({
+          actionTypeId,
+          status: ActionsCompletion.PARTIAL,
+        });
+
+        continue;
+      }
+
+      if (!this.isExecutableAction(action)) {
+        this.logger.warn(
+          `Rule "${this.taskInstance.params.alertId}" skipped scheduling action "${action.id}" because it is disabled`
+        );
+
+        continue;
+      }
+
+      ruleRunMetricsStore.incrementNumberOfTriggeredActions();
+      ruleRunMetricsStore.incrementNumberOfTriggeredActionsByConnectorType(actionTypeId);
+
+      if (summarizedAlerts && !isSystemAction(action)) {
+        const { actionsToEnqueueForExecution, actionsToLog } = await this.runSummarizedAction({
```

Comment:
Small refactor. I create one run*Action method for each possible execution and I move the logic inside the dedicated functions. The logic is the same as before.